src/test_l1_dsAudio.c

```diff
@@ -99,6 +100,106 @@ intptr_t null_handle = 0;
 
 /* Global flags to support features */
 static bool extendedEnumsSupported=false; //Default to not supported
+static char* device_type = NULL;
+static char* device_name = NULL;
+static int numports = 0;
+
+typedef struct  port_config {
+	uint16_t typeid;
+	uint16_t index;
+	char* port_name;
+	int no_of_supported_compression;
+	int supported_compressions[ARRAY_SIZE];
```

Comment:
Use the dsAUDIO_CMP_MAX instead of ARRAY_SIZE. Same for other parameters, instead of using ARRAY_SIZE use corresponding enum max value